src/socket_poller.cpp

```diff
@@ -464,117 +592,28 @@ int zmq::socket_poller_t::wait (zmq::socket_poller_t::event_t *events_, int n_ev
             signaler->recv ();
 
         //  Check for the events.
-        int found = 0;
-        for (items_t::iterator it = items.begin (); it != items.end () && found < n_events_; ++it) {
-
-            events_[found].socket = NULL;
-            events_[found].fd = 0;
-            events_[found].user_data = NULL;
-            events_[found].events = 0;
-
-            //  The poll item is a 0MQ socket. Retrieve pending events
-            //  using the ZMQ_EVENTS socket option.
-            if (it->socket) {
-                size_t events_size = sizeof (uint32_t);
-                uint32_t events;
-                if (it->socket->getsockopt (ZMQ_EVENTS, &events, &events_size) == -1) {
-                    return -1;
-                }
-
-                if (it->events & events) {
-                    events_[found].socket = it->socket;
-                    events_[found].user_data = it->user_data;
-                    events_[found].events = it->events & events;
-                    ++found;
-                }
-            }
-            //  Else, the poll item is a raw file descriptor, simply convert
-            //  the events to zmq_pollitem_t-style format.
-            else {
-                short revents = pollfds [it->pollfd_index].revents;
-                short events = 0;
-
-                if (revents & POLLIN)
-                    events |= ZMQ_POLLIN;
-                if (revents & POLLOUT)
-                    events |= ZMQ_POLLOUT;
-                if (revents & POLLPRI)
-                    events |= ZMQ_POLLPRI;
-                if (revents & ~(POLLIN | POLLOUT | POLLPRI))
-                    events |= ZMQ_POLLERR;
-
-                if (events) {
-                    events_[found].socket = NULL;
-                    events_[found].user_data = it->user_data;
-                    events_[found].fd = it->fd;
-                    events_[found].events = events;
-                    ++found;
-                }
-            }
-        }
+        int found = check_events(events_, n_events_);
         if (found) {
-            for (int i = found; i < n_events_; ++i) {
-                events_[i].socket = NULL;
-                events_[i].fd = 0;
-                events_[i].user_data = NULL;
-                events_[i].events = 0;
-            }
+            if (found > 0)
```

Comment:
found is always non-negative, so this is redundant with the condition above

Reply:
No, it isn't.

`zmq::socket_poller_t::check_events()` can return -1, as a way to bail out (as the old code was doing in the body of the function).

With this branch, you avoid calling the cleanup function which doesn't support values < 0.

Reply:
Ok, sorry I did not see that. I thought it was an artifact of the refactoring :)